### DISCUSSION TITLE
Init function without x-init

### DISCUSSION THREAD

**Discussion by skeemer** [alpinejs/alpine#752] [4. General]
It would be nice to not have to call x-init directly. I've seen discussions about this and even some about .component(), but that  depends on calling the function a specific name from what I've seen. I'm wondering if this would be a good strategy without having to extend the API much. What if when the return from a component function is an array, the first element is the scope and the second is the init function.

```
function MyComponent(extraStuff) {
  const newState = {
     //  scope stuff
  }
  function init() {
     // run stuff on init
  }
  return [newState, init];
}
```

**Comment by mblarsen** [comment]
It is on the [V3 roadmap](https://github.com/alpinejs/alpine/#v3-roadmap):

> Add Alpine.component('foo', {...}) (With magic __init() method)

**Reply by skeemer** [reply]
That just seems like overkill and a lot larger change to the API to add one small feature.

**Comment by KevinBatdorf** [comment]
You could write your own js script to add an auto init:
```js
Array.from(document.querySelectorAll('[x-data]')).forEach(alpine => {
  alpine.setAttribute('x-init',
    [
      alpine.getAttribute('x-init'),
      `typeof init == 'function' && init()`
    ].filter(Boolean).join(';')
  )
})
```
then
```html
<data x-data="{ something: true, init() { this.something = false } }"></div>
```

**Reply by skeemer** [reply]
Thank you! That saved me from some headaches with updating a component that now needs init.
